src/dto/rpc/CosmosRpcSuite.ts

```diff
@@ -0,0 +1,1600 @@
+  data: BlockData;
+  evidence: Evidence;
+  last_commit: LastCommit;
+}
+
+// Get latest block response interface
+export interface GetLatestBlockResponse {
+  block_id: BlockID;
+  block: Block;
+}
+
+export interface GetBlockByHeightRequest {
+  height: number;
+}
+
+// Block ID interface
+interface BlockID {
+  hash: string;
+  part_set_header: {
+    total: number;
+    hash: string;
+  };
+}
+
+// Block header interface
+interface BlockHeader {
+  version: {
+    block: string;
+    app: string;
+  };
+  chain_id: string;
+  height: string;
+  time: string;
+  last_block_id: BlockID;
+  last_commit_hash: string;
+  data_hash: string;
+  validators_hash: string;
+  next_validators_hash: string;
+  consensus_hash: string;
+  app_hash: string;
+  last_results_hash: string;
+  evidence_hash: string;
+  proposer_address: string;
+}
+
+// Evidence interface
+interface Evidence {
+  evidence: any[];
+}
+
+// Commit signature interface
+interface CommitSignature {
+  block_id_flag: string;
+  validator_address: string;
+  timestamp: string;
+  signature: string;
+}
+
+// Last commit interface
+interface LastCommit {
+  height: string;
+  round: number;
+  block_id: BlockID;
+  signatures: CommitSignature[];
+}
+
+// Get block by height response interface
+export interface GetBlockByHeightResponse {
+  block_id: BlockID;
+  block: Block;
+}
+
+// Protocol version interface
+interface ProtocolVersion {
+  p2p: string;
+  block: string;
+  app: string;
+}
+
+// Default node info interface
+interface DefaultNodeInfo {
+  protocol_version: ProtocolVersion;
+  default_node_id: string;
+  listen_addr: string;
+  network: string;
+  version: string;
+  channels: string;
+  moniker: string;
+  other: {
+    tx_index: string;
+    rpc_address: string;
+  };
+}
+
+// Build dependency interface
+interface BuildDependency {
+  path: string;
+  version: string;
+  sum: string;
+}
+
+// Application version interface
+interface ApplicationVersion {
+  name: string;
+  app_name: string;
+  version: string;
+  git_commit: string;
+  build_tags: string;
+  go_version: string;
+  build_deps: BuildDependency[];
+  cosmos_sdk_version: string;
+}
+
+// Get node info response interface
+export interface GetNodeInfoResponse {
+  default_node_info: DefaultNodeInfo;
+  application_version: ApplicationVersion;
+}
+
+export interface GetSyncingResponse {
+  syncing: boolean;
+}
+
+// Validator pub key interface
+interface ValidatorPubKey {
+  "@type": string;
+  key: string;
+}
+
+// Validator interface
+interface Validator {
+  address: string;
+  pub_key: ValidatorPubKey;
+  voting_power: string;
+  proposer_priority: string;
+}
+
+
+// Get latest validator set response interface
+export interface GetLatestValidatorSetResponse {
+  block_height: string;
+  validators: Validator[];
+  pagination: PaginationResponse;
+}
+
+export interface GetValidatorSetByHeightRequest extends PaginationRequest {
+  height: number;
+}
+
+// Get validator set by height response interface
```

Comment:
do we really need all these comments? looks like they are just duplicating interfaces names

Reply:
yes will remove this.

Reply:
Next time I will use https://github.com/drwpow/openapi-typescript or something like this

Reply:
Or this OpenApi3-TS